charts/consul/templates/cni-resourcequota.yaml

```
@@ -0,0 +1,18 @@
+{{- if .Values.global.imageK8s }}{{ fail "global.imageK8s is not a valid key, use global.imageK8S (note the capital 'S')" }}{{ end -}}
+{{- if (and (.Values.connectInject.cni.enabled) (not .Values.connectInject.transparentProxy.defaultEnabled)) }}{{ fail "connectInject.transparentProxy.defaultEnabled must be true if connectInject.cni.enabled is true" }}{{ end -}}
+{{- if (and .Values.connectInject.cni.enabled .Values.connectInject.transparentProxy.defaultEnabled) }}
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: consul-cni-resource-quota
```

Just double-checking the name:
```suggestion
  name: {{ template "consul.fullname" . }}-cni
```

---

One more thing, does it make sense to add the `metadata.Labels` field like the other resources?

```
  labels:
    app: {{ template "consul.name" . }}
    chart: {{ template "consul.chart" . }}
    heritage: {{ .Release.Service }}
    release: {{ .Release.Name }}
```

---

It sure does!  Thanks for catching that!